src/app/public/modules/split-view/split-view.component.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChild,
+  ElementRef,
+  Input,
+  OnDestroy,
+  OnInit
+} from '@angular/core';
+
+import {
+  animate,
+  state,
+  style,
+  transition,
+  trigger
+} from '@angular/animations';
+
+import {
+  SkyCoreAdapterService,
+  SkyMediaQueryService
+} from '@skyux/core';
+
+import {
+  Subject
+} from 'rxjs/Subject';
+
+import 'rxjs/add/operator/take';
+
+import 'rxjs/add/operator/takeUntil';
+
+import 'rxjs/add/operator/takeWhile';
+
+import 'rxjs/add/observable/fromEvent';
+
+import {
+  SkySplitViewMessage
+} from './types/split-view-message';
+
+import {
+  SkySplitViewMessageType
+} from './types/split-view-message-type';
+
+import {
+  SkySplitViewDrawerComponent
+} from './split-view-drawer.component';
+
+import {
+  SkySplitViewService
+} from './split-view.service';
+
+@Component({
+  selector: 'sky-split-view',
+  templateUrl: './split-view.component.html',
+  styleUrls: ['./split-view.component.scss'],
+  providers: [SkySplitViewService],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    trigger(
+      'blockAnimationOnLoad', [
+        transition(':enter', [])
+      ]
+    ),
+    trigger(
+      'drawerEnter', [
+        state('false', style({transform: 'translate(-100%)'})),
+        transition('* => true', animate('150ms ease-in'))
+      ]),
+    trigger(
+      'workspaceEnter', [
+        state('false', style({transform: 'translate(100%)'})),
+        transition('* => true', animate('150ms ease-in'))
+      ]
+    )
+  ]
+})
+export class SkySplitViewComponent implements OnInit, OnDestroy {
+  @Input()
+  public set backButtonText(value: string) {
+    if (value) {
+      this.splitViewService.updateBackButtonText(value);
+    }
+  }
+
+  @Input()
+  public messageStream = new Subject<SkySplitViewMessage>();
+
+  @ContentChild(SkySplitViewDrawerComponent)
+  public drawerComponent: SkySplitViewDrawerComponent;
+
+  public set drawerVisible(value: boolean) {
+    this._drawerVisible = value;
+    this.changeDetectorRef.markForCheck();
+  }
+
+  public get drawerVisible() {
+    return !this.isMobile || this._drawerVisible;
+  }
+
+  public isMobile = false;
+
+  public nextButtonDisabled = false;
+
+  public previousButtonDisabled = false;
+
+  public get workspaceVisible() {
+    return !this.isMobile || !this._drawerVisible;
+  }
+
+  private animationComplete = new Subject<void>();
+
+  private ngUnsubscribe = new Subject<void>();
+
+  private _drawerVisible = true;
+
+  constructor(
+    private changeDetectorRef: ChangeDetectorRef,
+    private coreAdapterService: SkyCoreAdapterService,
+    private elementRef: ElementRef,
```

Comment:
Personal preference - could we move `ElementRef` up one for alphabetical sorting?

Reply:
Wouldn't you think a developer would see the property name first? I feel like it takes less mental processing to look for the property name than it does to look for the name, but then also know the type. I really feel like alphabetizing the property name works better here.

Reply:
Oh. You're right. Ok. That's fine